File: backend/models/Product.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema({
    name: { type: String, required: true },
    description: { type: String, required: true },
    price: { type: Number, required: true },
    imageUrl: { type: String, required: true },
    rating: { type: Number, default: 4.5 },

    // UPDATED: Link to Category Model
    category: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Category',
        required: true
    },

    sizes: [{ name: String, price: Number }],
    addons: [{ name: String, price: Number }]
});

module.exports = mongoose.model('Product', productSchema);

File: app/src/main/java/com/shuham/urbaneats/domain/model/Product.kt
package com.shuham.urbaneats.domain.model

import kotlinx.serialization.Serializable

@Serializable // Needed for TypeConverter
data class Product(
    val id: String,
    val name: String,
    val description: String,
    val price: Double,
    val imageUrl: String,
    val rating: Double,
    val category: String,
    val isFavorite: Boolean = false,
    // NEW FIELDS
    val sizes: List<SizeOption> = emptyList(),
    val addons: List<AddonOption> = emptyList()
)